specification/Metadata/README.md

```diff
@@ -0,0 +1,446 @@
+
+#### Arrays
+
+Array values are encoded with varying array lengths and element sizes. Multiple arrays and array values are packed tightly in the same buffer.
+
+Variable-length arrays use an additional **array offset** buffer (`arrayOffset`). The `i-th` value in the array offset buffer is an element index — not a byte offset — identifying the beginning of the `i-th` array. String values within an array may have inconsistent lengths, requiring both array offset and **string offset** buffers (see: [Strings](#strings)).
+
+The data type used for offsets is defined by an **array offset type** (`arrayOffsetType`), which may be `UINT8`, `UINT16`, `UINT32`, or `UINT64`.
+
+If there are `N` arrays in the property array, the array offset buffer has `N + 1` elements. The first `N` of these point to the first element of an array within the property array, or within a string offset buffer for string component types. The last value points to a (non-existent) element immediately following the last array element.
+
+As a result, property value lookups for fixed- and variable-length arrays must compute an element's index differently. For each case below, the offset of an array element `i` within its binary storage is expressed in terms of entity ID `id` and element index `i`.
+
+| Array length | Array type                        | Offset type | Offset                                  |
+|--------------|-----------------------------------|-------------|-----------------------------------------|
+| variable     | `number[]`, `boolean[]`, `enum[]` | array index | `arrayOffset[id] + i`                   |
+| fixed        | `number[]`, `boolean[]`, `enum[]` | array index | `id * componentCount + i`               |
+| variable     | `string[]`                        | byte offset | `stringOffset[arrayOffset[id] + i]`     |
+| fixed        | `string[]`                        | byte offset | `stringOffset[id * componentCount + i]` |
+
+`VECN` and `MATN` types are treated as fixed-length numeric arrays.
+
+Each expression in the table above defines an index into the underlying property array. For a property array of `FLOAT32` components, index `3` would correspond to <u>_byte_</u> offset `3 * sizeof(FLOAT32) = 12` within that array. For an array of `BOOLEAN` components, offset `3` would correspond to <u>_bit_</u> offset `3`.
+
+> **Example:** Five variable-length arrays of UINT8 components, binary-encoded in a buffer. The associated property definition would be `type = "ARRAY"`, and `componentType = "UINT8"`, `componentCount = undefined` (variable-length).
+>
+> <img src="figures/array-of-ints.png"  alt="Variable-length array" width="640px">
+
+> **Example:** Two variable-length arrays of strings, binary-encoded in a buffer. The associated property definition would be `type = "ARRAY"`, `componentType = "STRING"`, `componentCount = undefined` (variable-length).
```

Comment:
Noticed one thing to clarify in this example:

```suggestion
> **Example:** Two variable-length arrays of strings, binary-encoded in a buffer. The associated property definition would be `type = "ARRAY"`, `componentType = "STRING"`, `componentCount = undefined` (variable-length). Observe that the last element of the array offset buffer points to the last element of the string offset buffer. This is because the last valid string offset is the next-to-last element of the string offset buffer.
```

Reply:
Ah good clarification! 👍